research/condist-fl/prepare_data.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import shutil
+from argparse import ArgumentParser
+from pathlib import Path
+
+import numpy as np
+import nibabel as nib
+from tqdm.contrib import tzip
+
+
+TASK_FILE_ID = {
+    "liver":    "LVR",
+    "spleen":   "SPL",
+    "pancreas": "PAN",
+    "kidney":   "KITS"
+}
+
+TASK_LABEL_MAP = {
+    "liver":    { 0: 0, 1: 1, 2: 2 },
+    "spleen":   { 0: 0, 1: 3 },
+    "pancreas": { 0: 0, 1: 4, 2: 5 },
+    "kidney":   { 0: 0, 1: 6, 2: 7 }
+}
+
+def map_values(data: np.ndarray, task: str) -> np.ndarray:
+    data = data.astype(np.uint8)
+
+    m = TASK_LABEL_MAP[task]
+    f = np.vectorize(lambda x: m[x])
+
+    return f(data).astype(np.uint8)
+
+def convert_msd_dataset(src: str, dst: str, task: str) -> None:
+    images = [str(f) for f in Path(src).glob("imagesTr/*.gz")]
```

Comment:
Assert `os.path.isdir(src)` and `len(images) > 0`

Reply:
I have added the assertions, thanks!